citizennet/purescript-ocelot
# FBCM-4154 Fix meridiem change

## What does this pull request do?

When a time value in the range of 12:00 AM to 12:59 AM (inclusive) is explicitly selected (either via dropdown or through the input field) the meridiem is changed from AM to PM on blur. This PR changes that behavior by doing two things:
1. Not attempting to guess the time if selected from a dropdown since only valid values are available to be selected
2. Allowing "a"/"A" to be a valid meridiem indicator when attempting to guess the time.

## How should this be manually tested?

1. Run `make ui-guide`
2. Click on the "Date Pickers" link
3. Select "12:00 AM" from the dropdown in one of the time pickers
4. Click off the focused input
- [x] The value should remain "12:00 AM"
5. Type a value in the problematic range (12:00 AM to 12:59 AM)
6. Click off the focused input
- [x] The value of the meridiem should remain "AM"
